### PR TITLE
Lowercase and replace white characters of the channel code in Channel…

### PR DESCRIPTION
**Pull request by bitbager**
… setup context

| Q               | A
| --------------- | ---
| Bug fix?        | yes |
| New feature?    | no |
| BC breaks?      | no |
| Related tickets | #8583 |
| License         | MIT |

### CODE REVIEW

File: src/Sylius/Behat/Context/Setup/ChannelContext.php [hunk 1]
```diff
@@ -114,7 +114,8 @@ public function storeOperatesOnASingleChannel($currencyCode = null)
      */
     public function theStoreOperatesOnAChannelNamed($channelIdentifier, $currencyCode = null)
     {
-        $defaultData = $this->defaultChannelFactory->create($channelIdentifier, $channelIdentifier, $currencyCode);
+        $channelCode = $this->resolveChannelCode($channelIdentifier);
```
Comment (Zales0123) [inline src/Sylius/Behat/Context/Setup/ChannelContext.php]: We can use `StringInflector::nameToLowercaseCode(string $value)` method instead.

File: src/Sylius/Behat/Context/Setup/ChannelContext.php [hunk 2]
```diff
@@ -114,7 +115,8 @@ public function storeOperatesOnASingleChannel($currencyCode = null)
      */
     public function theStoreOperatesOnAChannelNamed($channelIdentifier, $currencyCode = null)
     {
-        $defaultData = $this->defaultChannelFactory->create($channelIdentifier, $channelIdentifier, $currencyCode);
+        $channelCode = StringInflector::nameToLowercaseCode($channelIdentifier);
```
Comment (lchrusciel) [inline src/Sylius/Behat/Context/Setup/ChannelContext.php]: It is not used anywhere, so I would put it in one line. But that's just a suggestion

Comment (bitbager) [reply src/Sylius/Behat/Context/Setup/ChannelContext.php]: I'm doing it like this because IMHO it's more readable. Shall I change it?

**Comment by bitbager** [comment]
Next time I will 🎅

File: src/Sylius/Behat/Context/Setup/ChannelContext.php (hunk 2 above)
Comment (lchrusciel) [reply src/Sylius/Behat/Context/Setup/ChannelContext.php]: Not needed. Just a suggestion. It is not something I would kill for 🔪

**Comment by lchrusciel** [comment]
@bitbager it is good to go for me, but one scenario failed. Can you fix it?

File: src/Sylius/Behat/Context/Setup/ChannelContext.php (hunk 2 above)
Comment (pamil) [reply src/Sylius/Behat/Context/Setup/ChannelContext.php]: Since it has introduced `$channelCode`, I would rename `$channelIdentifier` to `$channelName`

**Comment by bitbager** [comment]
@lchrusciel, sure 🙂

**Comment by bitbager** [comment]
Ok, it's still failing. Looks like the bug appears in more scenarios. I will try my best to fix it before midnight.

File: src/Sylius/Behat/Context/Ui/Admin/DashboardContext.php [hunk 3]
```diff
@@ -44,11 +45,11 @@ public function iOpenAdministrationDashboard()
...
 
     /**
-     * @When I open administration dashboard for :code channel
+     * @When I open administration dashboard for :name channel
      */
-    public function iOpenAdministrationDashboardForChannel($code)
+    public function iOpenAdministrationDashboardForChannel($name)
     {
-        $this->dashboardPage->open(['channel' => $code]);
+        $this->dashboardPage->open(['channel' => StringInflector::nameToCode($name)]);
```
Comment (pamil) [inline src/Sylius/Behat/Context/Ui/Admin/DashboardContext.php]: Shouldn't it be `nameToLowercaseCode()`?

**Comment by bitbager** [comment]
@pamil, looks like a pass 🎉

**Comment by pjedrzejewski** [comment]
Thank you Mikołaj! 👍

**Pull request merged**